src/org/labkey/test/components/react/ToggleButton.java

```diff
@@ -32,31 +32,57 @@ public WebDriver getDriver()
     public ToggleButton set(boolean enabled)
     {
         String desiredState = enabled ? "enabled" : "disabled";
-        if (get() != enabled)
-            getComponentElement().click();
-        WebDriverWrapper.waitFor(()-> get() == enabled,
+        if (enabled && isDisabled())
+        {
+            if (hasButtons()) selectFirst();
+            else getComponentElement().click();
+        }
+        else if (!enabled && isEnabled())
+        {
+            if (hasButtons()) selectSecond();
+            else getComponentElement().click();
+        }
```

Comment:
This should have some failure state if `isEnabled` and `isDisabled` are both false. That would indicate that the test component is out of sync with the product, right?
```suggestion
        else if (!enabled && isEnabled())
        {
            if (hasButtons()) selectSecond();
            else getComponentElement().click();
        }
        else
        {
            throw new IllegalStateException("Unable to determine toggle state: " + getComponentElement());
        }
```

Reply:
Yes. Added.